src/nccl_ofi_scheduler.c

```diff
@@ -22,10 +22,27 @@ static inline size_t sizeof_schedule(int num_rails)
 		+ num_rails * sizeof(nccl_net_ofi_xfer_info_t);
 }
 
-void nccl_net_ofi_set_multiplexing_schedule(size_t size, int num_rails,
+int set_schedule_by_threshold(nccl_net_ofi_threshold_scheduler_t *scheduler,
+					    size_t size,
+					    int num_rails,
 					    size_t align,
 					    nccl_net_ofi_schedule_t *schedule)
 {
+	int ret = 0;
+	int num_stripes = (int) NCCL_OFI_MIN(NCCL_OFI_DIV_CEIL(size, scheduler->min_stripe_size), num_rails);
+	if (OFI_UNLIKELY(num_rails == 0) || OFI_UNLIKELY(num_stripes == 0)) return -1;
+
+	if (OFI_UNLIKELY(num_stripes > num_rails)) return -1;
+
+	int rail_id;
+	nccl_net_ofi_mutex_lock(&scheduler->rr_lock);
+
+	/* Retieve and increment multiplex-round-robin counter; wrap around if required */
+	rail_id = scheduler->rr_counter;
```

Comment:
But we use rail_id immediately right . 
The next_rail_id is actually incremented in the scheduler->rr_counter  shown below ?
`scheduler->rr_counter = (scheduler->rr_counter + num_stripes) % num_rails;`

Reply:
It's really hard to figure out what you're doing with this variable up front.  The names should tell you that.  You don't have to like my suggestion, but you do need to find a name that indicates what you're doing with the data.  I picked next_rail_id, because it's the next rail id that you'll use for assignment.  But feel free to come up with a different name, as long as it describes what the variable is there for.

Reply:
Sure.  I have instead changed the logic such that we have curr_rail_id which we fetched from the rr_scheduler and next_rail_id which is incremented by the (num_stripes % num_rails) and is assigned back to rr_counter.
```
curr_rail_id = scheduler->rr_counter;
next_rail_id = (curr_rail_id + num_stripes) % num_rails;
```